src/mlir/generator.cc

```diff
@@ -223,22 +217,44 @@ namespace mlir::verona
     // Lower body
     auto body = getFunctionBody(ast);
     auto last = parseNode(body.lock());
+    if (auto err = last.takeError())
+      return std::move(err);
 
     // Return last value
-    if (last && last.getType() != retTy)
+    if (*last && last->getType() != retTy)
     {
       // Cast type (we trust the ast)
-      last = genOperation(last.getLoc(), "verona.cast", {last}, retTy);
+      last = genOperation(last->getLoc(), "verona.cast", {*last}, retTy);
     }
     else
     {
       last = genOperation(getLocation(ast), "verona.none", {}, retTy);
     }
-    builder.create<mlir::ReturnOp>(getLocation(ast), last);
+    if (auto err = last.takeError())
+      return std::move(err);
+    builder.create<mlir::ReturnOp>(getLocation(ast), *last);
 
     return func;
   }
 
+  mlir::Type Generator::parseType(const ::ast::Ast& ast)
```

Comment:
Why did you move the location of this function?

Reply:
to match the order in the header file.